plugins/teststeps/awsfileupload/awsFileUpload.go

```diff
@@ -0,0 +1,299 @@
+	// validate path and filename
+	ts.path = params.GetOne("path")
+	if ts.path.IsEmpty() {
+		return fmt.Errorf("invalid or missing 'path' parameter, must be exactly one string")
+	}
+	ts.filename = params.GetOne("filename")
+	if ts.filename.IsEmpty() {
+		return fmt.Errorf("invalid or missing 'path' parameter, must be exactly one string")
+	}
+
+	// Retrieving parameter as string
+	// validate s3region
+	param := params.GetOne("s3region")
+	if param.IsEmpty() {
+		return fmt.Errorf("missing 's3region' parameter, must be exactly one string")
+	}
+	ts.s3region = param.String()
+	// validate s3bucket
+	param = params.GetOne("s3bucket")
+	if param.IsEmpty() {
+		return fmt.Errorf("missing 's3bucket' parameter, must be exactly one string")
+	}
+	ts.s3bucket = param.String()
+	// validate s3path
+	param = params.GetOne("s3path")
+	if param.IsEmpty() {
+		return fmt.Errorf("missing 's3path' parameter, must be exactly one string")
+	}
+	ts.s3path = param.String()
+	// retrieve s3credfile
+	ts.s3credfile = params.GetOne("s3credfile").String()
+	// retrieve s3credprofile
+	ts.s3credprofile = params.GetOne("s3credprofile").String()
+
+	// Retrieving parameter as bool
+	// validate emit_urltoreport
+	param = params.GetOne("emit_urltoreport")
+	if !param.IsEmpty() {
+		v, err := strconv.ParseBool(param.String())
+		if err != nil {
+			return fmt.Errorf("invalid non-boolean `emit_urltoreport` parameter: %v", err)
+		}
+		ts.emitURLtoReport = v
+	}
+	// validate compress
+	param = params.GetOne("compress")
+	if !param.IsEmpty() {
+		v, err := strconv.ParseBool(param.String())
+		if err != nil {
+			return fmt.Errorf("invalid non-boolean `emit_urltoreport` parameter: %v", err)
+		}
+		ts.compress = v
+	}
+	return nil
+}
+
+// ValidateParameters validates the parameters associated to the TestStep
+func (ts *FileUpload) ValidateParameters(_ xcontext.Context, params test.TestStepParameters) error {
+	return nil
+}
+
+// New initializes and returns a new awsFileUpload test step.
+func New() test.TestStep {
+	return &FileUpload{}
+}
+
+// Load returns the name, factory and events which are needed to register the step.
+func Load() (string, test.TestStepFactory, []event.Name) {
+	return Name, New, Events
+}
+
+// createTarArchive creates compressed data writer and invokes addFileToArchive
+func createTarArchive(file string, buf bytes.Buffer) (bytes.Buffer, error) {
+	// Create gzip and tar writers
+	gzwriter := gzip.NewWriter(&buf)
+	defer gzwriter.Close()
+	tarwriter := tar.NewWriter(gzwriter)
+	defer tarwriter.Close()
+	// Write file into tar archive
+	err := addFileToArchive(tarwriter, file)
+	if err != nil {
+		return buf, err
+	}
+	return buf, nil
+}
+
+// addFileToArchive takes the data and writes it into the tar archive
+func addFileToArchive(tarwriter *tar.Writer, filename string) error {
+	// Open the file which shall be written into the tar archive
+	file, err := os.Open(filename)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
```

Comment:
`file.Close()` should be checked, it can fail in significant ways (I can see Tobias chuckle at this comment :D ). For example:
```
defer func() {
    if err := file.Close(); err != nil {
         ctx.Logger().Warnf("failed to close file '%s': %w", filename, err)
    }
}
```

Reply:
Aaaaaaah!